pkg/asset/rhcos/image.go

```diff
@@ -77,16 +77,14 @@ func osImage(config *types.InstallConfig) (string, error) {
 		osimage, err = rhcos.GCP(ctx)
 	case libvirt.Name:
 		osimage, err = rhcos.QEMU(ctx)
-	case openstack.Name:
-		osimage = "rhcos"
 	case azure.Name:
 		osimage, err = rhcos.VHD(ctx)
 	case baremetal.Name:
 		// Note that baremetal IPI currently uses the OpenStack image
 		// because this contains the necessary ironic config drive
 		// ignition support, which isn't enabled in the UPI BM images
 		osimage, err = rhcos.OpenStack(ctx)
-	case none.Name, vsphere.Name:
+	case none.Name, openstack.Name, vsphere.Name:
```

Comment:
Shouldn't it be `clusterID.InfraID + "-rhcos"` for the openstack case so that you don't have to check for empty value each time?

This isn't immediately obvious why we do things differently for OpenStack. Could you also explain why not using `osimage, err = rhcos.OpenStack(ctx)` ? I'm thinking it may be so that we can destroy cluster in a tenant without impacting other clusters in the same tenant. Would be nice to add a comment.

Reply:
clusterID is not generated yet here, so we don't know its infra id. 

We do things differently for OpenStack, because image creation is not idempotent there. When we ask Terraform to create "rhcos" image in Glance 5 times it will create 5 different images, which is not true for other platforms.

rhcos.OpenStack(ctx) returns a url where we can find our binary file (like https://releases-art-rhcos.svc.ci.openshift.org/art/storage/releases/rhcos-4.3/43.80.20191002.1/x86_64/rhcos-43.80.20191002.1-openstack.x86_64.qcow2) But we expect osImage to return a Glance image name, "rhcos" for instance. So we can't use this function here directly. Instead, image names are generated later and there we know if this string is empty, then Terraform needs to create a new image "<InfraID>-rhcos".

Reply:
Is anything preventing us from generating the clusterID earlier then?
My point being, we should never have the image name being an empty string. It is either generated based on the clusterID or set via `OPENSHIFT_INSTALL_OS_IMAGE_OVERRIDE`.